APCAppCore/APCAppCore/Library/DataArchiverAndCollector/APCPassiveDataCollector.m

```diff
@@ -309,20 +218,31 @@ - (void) APCDataTracker:(APCDataTracker *)tracker hasNewData:(NSArray *)dataArra
 - (void) checkIfDataNeedsToBeFlushed:(APCDataTracker*) tracker
 {
     //Check for size
-    NSString * csvFilePath = [tracker.folder stringByAppendingPathComponent:kCSVFilename];
-    NSError * error;
+    NSString *csvFilePath = [tracker.folder stringByAppendingPathComponent:kCSVFilename];
+    NSError *error;
     NSDictionary *fileDictionary = [[NSFileManager defaultManager] attributesOfItemAtPath:csvFilePath error:&error];
-    APCLogError2(error);
-    unsigned long long filesize = [fileDictionary fileSize];
-    if (filesize >= tracker.sizeThreshold) {
-        [self flush:tracker];
+    
+    if (!fileDictionary) {
+        APCLogError2(error);
+    } else {
+        unsigned long long filesize = [fileDictionary fileSize];
+        
+        if (filesize >= tracker.sizeThreshold) {
+            [self flush:tracker];
+        }
     }
     
     //Check for start date
-    NSDictionary * dictionary = tracker.infoDictionary;
-    NSString * startDateString = dictionary[kStartDateKey];
-    NSDate * startDate = [self datefromDateString:startDateString];
-    if ([[NSDate date] timeIntervalSinceDate:startDate] >= tracker.stalenessInterval) {
+    NSDictionary *dictionary = tracker.infoDictionary;
+    NSString *startDateString = dictionary[kStartDateKey];
+    NSDate *startDate = [self datefromDateString:startDateString];
+    
+    if (!startDate) {
+        startDate = [NSDate date];
+    }
+    
+    if ([[NSDate date] timeIntervalSinceDate:startDate] >= tracker.stalenessInterval)
+    {
         [self flush:tracker];
```

Comment:
I have updated the method to trigger flush only once, when either of the two (file size or time limit) conditions are met. The changes will be reflected on the pull request shortly.